Library/Homebrew/version.rb

```diff
@@ -436,7 +436,7 @@ def self._parse(spec)
     return m.captures.first unless m.nil?
 
     # e.g. https://www.openssl.org/source/openssl-0.9.8s.tar.gz
-    m = /-v?([^-]+)/.match(stem)
+    m = /-v?(\d+(?:\.\d+)*[^-]*)/.match(stem)
```

Comment:
If you make this `m = /-v?(\d+[^-]+)/.match(stem)`, it should correctly fall through to the code designed to detect version numbers from this sort of url.